### PR TITLE
chore(Topology): minimize some imports

### PR DESCRIPTION
**Pull request by grunweg**
For faster build times and clearer dependencies. No attempt at being exhaustive.
- [x] depends on: #7628

--------------

Rebasing my PR over #7628, I noticed some extraneous imports in `Compactness/Compact.lean` and followed from there.
Found using #minimize_imports (and checking things still build) - that command is really helpful!

### CODE REVIEW

**Comment by grunweg** [comment]
This PR should rebase cleanly; after that, should be straightforward to review.

File: Mathlib/Topology/MetricSpace/Lipschitz.lean [hunk 1]
```diff
@@ -3,11 +3,10 @@ Copyright (c) 2018 Rohan Mitta. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Rohan Mitta, Kevin Buzzard, Alistair Tucker, Johannes Hölzl, Yury Kudryashov
 -/
-import Mathlib.Logic.Function.Iterate
+import Mathlib.Topology.Bornology.Hom
```
Comment (jcommelin) [inline Mathlib/Topology/MetricSpace/Lipschitz.lean:6]: Is there a reason that you moved this import to the top of the list?

File: Mathlib/Topology/Connected.lean [hunk 2]
```diff
@@ -3,10 +3,9 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro, Yury Kudryashov
 -/
-import Mathlib.Data.Set.BoolIndicator
-import Mathlib.Order.SuccPred.Relation
 import Mathlib.Topology.Clopen
 import Mathlib.Topology.Irreducible
+import Mathlib.Order.SuccPred.Relation
```
Comment (jcommelin) [inline Mathlib/Topology/Connected.lean:8]: Is there a reason you moved this import down?

File: Mathlib/Topology/Compactness/Compact.lean [hunk 3]
```diff
@@ -3,14 +3,10 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
...
-import Mathlib.Order.Filter.Pi
-import Mathlib.Topology.Bases
-import Mathlib.Data.Finset.Order
 import Mathlib.Data.Set.Accumulate
-import Mathlib.Data.Set.BoolIndicator
-import Mathlib.Topology.Bornology.Basic
+import Mathlib.Topology.Bases
 import Mathlib.Topology.LocallyFinite
-import Mathlib.Order.Minimal
+import Mathlib.Topology.Bornology.Basic
```
Comment (jcommelin) [inline Mathlib/Topology/Compactness/Compact.lean:9]: Is there a reason you moved this import down?

File: Mathlib/Topology/Compactness/Compact.lean [hunk 4]
```diff
@@ -3,14 +3,10 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro, Yury Kudryashov
 -/
-import Mathlib.Order.Filter.Pi
-import Mathlib.Topology.Bases
-import Mathlib.Data.Finset.Order
 import Mathlib.Data.Set.Accumulate
-import Mathlib.Data.Set.BoolIndicator
-import Mathlib.Topology.Bornology.Basic
+import Mathlib.Topology.Bases
```
Comment (jcommelin) [inline Mathlib/Topology/Compactness/Compact.lean:7]: Is there a reason you moved this import down?

File: Mathlib/Topology/MetricSpace/Lipschitz.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Topology/MetricSpace/Lipschitz.lean:6]: Now they're in alphabetical.
I understand import order is sometimes important, but when it's not, alphabetical makes the most sense to me.

File: Mathlib/Topology/Connected.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/Topology/Connected.lean:8]: Same here.

File: Mathlib/Topology/Compactness/Compact.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/Topology/Compactness/Compact.lean:9]: And here.

File: Mathlib/Topology/Compactness/Compact.lean (hunk 4 above)
Comment (grunweg) [reply Mathlib/Topology/Compactness/Compact.lean:7]: And here as well.

**Comment by grunweg** [comment]
awaiting-review

File: Mathlib/Topology/MetricSpace/Lipschitz.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/Topology/MetricSpace/Lipschitz.lean:6]: How is this alphabetical? Surely Bornology comes after Algebra?

Comment (grunweg) [reply Mathlib/Topology/MetricSpace/Lipschitz.lean]: :facepalm: Indeed, reverted. Too early in the morning...

**Comment by grunweg** [comment]
I have reverted the reorderings (except for one, where it's actually more alphabetical). Thanks for catching this.

**Comment by grunweg** [comment]
By a branch mixup, I re-did this as #7629. Closing in favour of that.

**Pull request closed without merging**